### PR TITLE
Update to where syntax for math files

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

**Comment by ararslan** [comment]
It would be nice to use the brackets consistently, i.e. `where S<:T` or `where {S<:T}` rather than both.

**Comment by musm** [comment]
brackets are only used in short form declarations

**Comment by ararslan** [comment]
Brackets can be used in both contexts. As an example, [brackets in long form](https://github.com/JuliaLang/julia/blob/333ff80e852aa20e9568e47d395ded82ad73f8d0/test/keywordargs.jl#L263), [no brackets in short form](https://github.com/JuliaLang/julia/blob/94417e46a16e26a51cc1171889659a192c15e6d6/test/inference.jl#L80).

**Comment by musm** [comment]
Yes I'm well aware. I was stating that I have stuck to the convention of using brackets in short form declarations since it's less confusing (e.g. `f(x::T) where T = 1`)  and have not used brackets in long form declarations since there's not source of confusion in this context.

File: base/float.jl [hunk 1]
```diff
@@ -519,7 +519,7 @@ abs(x::Float64) = abs_float(x)
 Test whether a floating point number is not a number (NaN).
 """
 isnan(x::AbstractFloat) = x != x
-isnan(x::Float16)    = reinterpret(UInt16,x)&0x7fff  > 0x7c00
+isnan(x::Float16) = reinterpret(UInt16,x)&0x7fff  > 0x7c00
```
Comment (alyst) [inline base/float.jl]: also extra space before comparison

**Review by Sacha0 — APPROVED** [review]
lgtm! :)

**Pull request merged**